docs/release.md

```diff
@@ -0,0 +1,68 @@
+# OSA Release process
+
+OSA Release process consist of multiple dimensions:
+* Plugin code release
+* Config release
+* VM image release
+
+Update might be one or any combination of all dimensions.
+Plugin code release is a new branch and considered a major release. Any VM image or config change is a minor release.
+
+# Versioning
+
+## Major releases
+
+Major release creation flow:
+1. Create a branch with a plugin code you wish to release.
+2. Commit to update `PluginConfig` referencing release image versions.
+3. Git tag major release version (`v3.0`)
+4. Build release images (sync, metricsbridge, etc)
+5. Make sure VM image is released and is available to be used by plugin code.
+
+## Minor releases
+
+If there is a need to update already released code - minor release is done.
+This might happen if we make a mistake, want to update configuration, or not code related fix.
+
+Minor release creation flow:
+1. Add commit to release branch with required changes.
+2. Git tag minor version (`v3.1`)
+3. Build release images (sync, metricsbridge, etc)
+
+Branching model:
+```
+Master-branch--*----------*-----------------------
+                \          \
+                 \-Release-v3------T(v3.0)----
+                             \-Release-v4----T(v4.0)-----T(v4.1)-
+```
+Note: Branch creation doesn't correspond to release, tags does.
```

Comment:
There is really no change between e.g. "tagging v3.0" and "tagging v3.1", as both may require:
* code changes
* config changes
* doc changes

The _only_ difference lies in cutting a new branch "release-v3" which is a pre-requisite for "tag-v3.0"

Reply:
I think if we release new code (if this is not typo fixing or super small logic change) we will do a new release. But we need to be cautious so we would not get ourselves into "upgrade hell"